streams/src/main/java/org/apache/kafka/streams/state/internals/RocksDBSegmentedBytesStore.java

```diff
@@ -138,5 +157,27 @@ public boolean persistent() {
     public boolean isOpen() {
         return open;
     }
+    
+    private static class TimeRangeSegmentIterator extends SegmentIterator { 
```

Comment:
Yeah I think without the key there is not too much we can do than scanning the whole store unfortunately.

About the code structure: instead of keeping this private static class in `RocksDBSegmentedBytesStore`, I'd suggest we make it a separate class in `org.apache.kafka.streams.state.internals`.